src/models/streaming/BufferedDuplex.ts

```diff
@@ -12,28 +12,28 @@ export class BufferedDuplex extends Duplex {
      * @param output Buffer for the Readable side of the stream.
      * @param options Options to be passed through to the superclass.
      */
-    constructor (input: PassThrough, output: PassThrough, options?: any) {
+    constructor (input: PassThrough, output: PassThrough, options?: DuplexOptions) {
         super(options);
         this.input = input;
         this.output = output;
         this.setupInput();
         this.setupOutput();
     }
 
-    public _write(chunk: any, encoding: string, callback: any): boolean {
+    public _write(chunk: any, encoding: string, callback: (error?: Error | null) => void): boolean {
```

Comment:
should type `error?: Error | null` lead to `Error | null | undefined`?
and what is the purpose of nullable error? is it for backward compatibility?

Reply:
Yeah this is the same as the `Duplex`'s typings in `@types/node`
```
// Note: Duplex extends both Readable and Writable.
class Duplex extends Readable implements Writable {
    writable: boolean;
    readonly writableHighWaterMark: number;
    readonly writableLength: number;
    constructor(opts?: DuplexOptions);
    _write(chunk: any, encoding: string, callback: (error?: Error | null) => void): void;
```